### PR TITLE
Use random default port instead of a fix one

### PR DESCRIPTION
**Pull request by magic282**
# Before submitting

- [y] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [y] Did you read the [contributor guideline](https://github.com/williamFalcon/pytorch-lightning/blob/master/.github/CONTRIBUTING.md)?
- [y] Did you make sure to update the docs?   
- [n] Did you write any new necessary tests?  

## What does this PR do?
Fixes https://github.com/williamFalcon/pytorch-lightning/issues/485.

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/core/lightning.py [hunk 1]
```diff
@@ -603,7 +603,8 @@ def init_ddp_connection(self):
             default_port = int(default_port) + 15000
 
         except Exception as e:
```
Comment (Borda) [inline pytorch_lightning/core/lightning.py]: pls, drop the `as e`

File: pytorch_lightning/core/lightning.py [hunk 2]
```diff
@@ -603,7 +603,8 @@ def init_ddp_connection(self):
...
-            default_port = 12910
+            import random
+            default_port = random.randint(15000,65535)
```
Comment (Borda) [inline pytorch_lightning/core/lightning.py]: user `numpy.random.randint()`

**Comment by williamFalcon** [comment]
actually, this won’t work.
All processes need the same port. making the port random will break ddp

**Comment by Borda** [comment]
so what about random port which will be the save over the package?

**Comment by williamFalcon** [comment]
how? ddp starts a process independently for each GPU, and you can’t guarantee the same number.

**Comment by Borda** [comment]
You would have to set it in init, but I need to have look at it in detail...

**Comment by williamFalcon** [comment]
can’t set in init because the processes all run at different times without guarantee of the same start time

**Comment by magic282** [comment]
How about hashing the python cmd args to a get a fix random number for all processes as a workaround?

**Comment by Borda** [comment]
well if you set `numpy.random.seed(42)`then all random numbers are the same, it is commonly used for tests with random matrixes to have every time/run the same random numbers...

**Comment by williamFalcon** [comment]
random seed is standard for anything ML. that doesn’t solve the problem though.  hashing the command might be a good option

**Pull request closed without merging**